### ISSUE TITLE
Troubleshooting Section for Quickstarts

### ISSUE THREAD

**Issue by Francesco-Lanciana** [thegooddocsproject/templates#100]
What do people think of advocating for a troubleshooting section inside a quickstart if there are common problems that they are aware of?

An example of this is [GitHub Apps Quickstart Guides](https://developer.github.com/apps/quickstart-guides/). A counterexample is [Stripes API Quickstart](https://stripe.com/docs/development) where they seem to assume a certain level of competency with your language of choice.

**Comment by shreyas1599** [comment]
@Francesco-Lanciana Even the Github Apps Quickstart Guides lists a set of [pre-requisites](https://developer.github.com/apps/quickstart-guides/setting-up-your-development-environment/#prerequisites) before reading the quickstart guide. I think adding a troubleshooting section makes the quickstart slightly more accessible to the beginner user? So, I personally agree with GitHub's approach and think it should be added.

**Comment by jaredmorgs** [comment]
I'm -1 to having a troubleshooting section inside the Quickstart.

Prerequisites are something I would include, which, if followed, would _take the place of_ troubleshooting. The idea is that if you complete the prerequisites, there should be no need to troubleshoot. 

Quickstarts for me are the one place in your docs (and the product) where you need to put in the extra effort, so the user's first experience with your product is 100% positive. 

You should solve any significant issues with the whole quickstart process, so it does not require any troubleshooting from the end-user. 
And if you do find end-users contacting you about dead-ends or areas in the procedure that might suggest a troubleshooting section, you should resist the temptation and just fix the problems either in the quickstart procedures or the product itself.

**Comment by shreyas1599** [comment]
@jaredmorgs Right. Makes more sense. I'm still learning about best practices and can say I've picked a few along the way reading through discussions in the issues and the repo in general. I was specifically addressing the issue of competency vs non-competency and hence suggested including it but your view takes care of that by putting the onus on the product creators. +1 to this viewpoint.
